podpac/core/node.py

```diff
@@ -657,7 +638,7 @@ def cached_function():
                     out = self.get_cache(key)
                 except NodeException:
                     out = func(self)
-                    self.put_cache(out, key, raise_no_cache_exception=raise_no_cache_exception)
```

Comment:
What happens now without this flag?

Reply:
There is always a cache_ctrl, but it may be empty. If the cache_ctrl is empty, then `cache_ctrl.put` will do nothing.

I think if a node requires a cache, it should check that the cache_ctrl is not empty, probably when it is created. Truthfully, if a node requires a cache, that's not what a cache is. We're conflating a local data store with a cache, I guess because it is convenient. If a node wants to use a a disk store, another way it could do that would be to instantiate a DiskCacheStore object directly and then use it, rather than (ab)using the cache. Really, the DiskCacheStore is misnamed, it should just be a DiskStore, which the caching uses and other things can also use.

But I guess I could put `raise_no_cache_exception` back in and make the default `False`.

(I removed cache_ctrl=None because it was redundant. An empty CacheCtrl is the same: if a Node wanted to raise an exception if there was no cache ctrl, it would also want to raise an exception if the cache_ctrl was empty, right?)